### PR TITLE
fix(atomic): use correct delimiting character for category facet headless controller

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-2036

### CODE REVIEW

**Review by btaillon — APPROVED** [review]

**Review by btaillon — CHANGES_REQUESTED** [review]
isn't this breaking?

File: packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx [hunk 1]
```diff
@@ -224,7 +224,9 @@ export class AtomicCategoryFacet
       sortCriteria: this.sortCriteria,
       facetSearch: {numberOfValues: this.numberOfValues},
       basePath: this.basePath
-        ? this.basePath.split(',').map((value) => value.trim())
+        ? this.basePath
+            .split(this.delimitingCharacter)
+            .map((value) => value.trim())
```
Comment (btaillon) [inline packages/atomic/src/components/search/facets/atomic-category-facet/atomic-category-facet.tsx:229]: It's pretty intuitive to me that the prop is split by commas even if that's ont the delimiting character 🤔  I'm curious what Stencil's array props do.

Either way, we should update the prop's description to reflect whichever way we go.

**Comment by ThibodeauJF** [comment]
Unfortunately yes it's breaking, we need to do something "uglier" until the next major. Maybe have a way to escape the commas in the path value

**Comment by olamothe** [comment]
@btaillon @ThibodeauJF : Ready to re-review.

Old format (comma separated) will still work properly, but will not print a warning to the console about deprecation, and the need to move to a JSON string representation.

If you like this approach, after merging this PR I can also do the job of going through other comma separated list properties, and move them to this new format.

File: packages/atomic/src/utils/props-utils.ts [hunk 2]
```diff
@@ -30,6 +38,56 @@ export function MapProp(opts?: MapPropOptions) {
...
+    opts.deprecationWarning
+      ? console.warn(
+          `${msg} This will be enforced in the next major version`,
+          other
+        )
+      : console.error(msg, other);
+
+  return (component: ComponentInterface, variableName: string) => {
+    const {componentWillLoad} = component;
+    const attributeWithBackets = camelToKebab(variableName);
```
Comment (btaillon) [inline packages/atomic/src/utils/props-utils.ts:52]: Did you mean to rename this? There's no more brackets.

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/utils/props-utils.ts (hunk 2 above)
Comment (olamothe) [reply packages/atomic/src/utils/props-utils.ts:52]: I removed the bracket annotation, yes

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**